compiler/llvm/clangUtil.cpp

```diff
@@ -2966,6 +2966,70 @@ getCGArgInfo(const clang::CodeGen::CGFunctionInfo* CGI, int curCArg)
   return argInfo;
 }
 
+static unsigned helpGetCTypeAlignment(const clang::QualType& qType) {
+  GenInfo* info = gGenInfo;
+  INT_ASSERT(info);
+  ClangInfo* clangInfo = info->clangInfo;
+  INT_ASSERT(clangInfo);
+
+  unsigned alignInBits = clangInfo->Ctx->getTypeAlignIfKnown(qType);
+
+  unsigned alignInBytes = alignInBits / 8;
+  // round it up to a power of 2
+  unsigned rounded = 1;
+  while (rounded < alignInBytes) rounded *= 2;
+
+  return rounded;
+}
+
+static unsigned helpGetCTypeAlignment(const clang::TypeDecl* td) {
+  QualType qType;
+
+  if (const TypedefNameDecl* tnd = dyn_cast<TypedefNameDecl>(td)) {
+    qType = tnd->getCanonicalDecl()->getUnderlyingType();
+  } else if (const EnumDecl* ed = dyn_cast<EnumDecl>(td)) {
+    qType = ed->getCanonicalDecl()->getIntegerType();
+    // could also use getPromotionType()
+    //could also do:
+    //  qType =
+    //   tnd->getCanonicalDecl()->getTypeForDecl()->getCanonicalTypeInternal();
```

Comment:
Did you leave these intentionally? I am not sure if it is helpful or not.

Reply:
I'll remove them.